docs/reference-guide/modules/commands/pages/command-handlers.adoc

```diff
@@ -1,249 +1,926 @@
 }
+----
+
+<1> `@EntityCreator` on a constructor that takes only the identifier creates an empty instance.
+Axon will use this to create an instance when a command arrives for an entity with no events.
+
+<2> An instance command handler can handle the creation command on the empty instance.
 
+**Pattern 2: Creation from origin event**
+
+This pattern creates an entity instance from the first event.
+Axon uses this when:
+
+* Loading an entity from the event store
+* The first event in the stream is used to create the initial instance
+* Subsequent events are then applied to update the instance
+
+[source,java]
 ----
+@EventSourced
+public class GiftCard {
 
-The `cardId` present in both commands is the reference to a `GiftCard` instance and thus is annotated with the `@TargetAggregateIdentifier` annotation.
-Commands that create an Aggregate instance do not need to identify the target aggregate identifier, as there is no Aggregate in existence yet.
-It is nonetheless recommended for consistency to annotate the Aggregate Identifier on them as well.
+    private final String id;
+    private final int initialAmount;
 
-If you prefer to use another mechanism for routing commands, the behavior can be overridden by supplying a custom `CommandTargetResolver`.
-This class should return the Aggregate Identifier and expected version (if any) based on a given command.
+    @EntityCreator // <1>
+    public GiftCard(CardIssuedEvent event, @InjectEntityId String id) {
+        this.id = event.cardId(); // or use the injected id parameter
+        this.initialAmount = event.amount();
+    }
+
+    @CommandHandler // <2>
+    public static String issue(IssueCardCommand cmd, EventAppender eventAppender) {
+        eventAppender.append(new CardIssuedEvent(cmd.cardId(), cmd.amount()));
+        return cmd.cardId();
+    }
+}
+----
+
+<1> `@EntityCreator` on a constructor that takes an event parameter.
+Axon uses the first event in the stream to create the initial instance.
+
+<2> A static command handler must be used for the creation command, since no instance exists to handle it yet.
+Once the event is published, future loads will use the `@EntityCreator` to create the instance from that event.
 
 [TIP]
-.Aggregate Creation Command Handlers
 ====
-When the `@CommandHandler` annotation is placed on an aggregate's constructor, the respective command will create a new instance of that aggregate and add it to the repository.
-Those commands do not require targeting a specific aggregate instance.
-Therefore, those commands need neither the `@TargetAggregateIdentifier` nor the `@TargetAggregateVersion` annotation.
-Furthermore, a custom `CommandTargetResolver` will not be invoked for these commands.
+Use `@InjectEntityId` to disambiguate the entity identifier from the event payload.
+Without this annotation, the first parameter is assumed to be the event payload.
 ====
 
-=== Business logic and state changes
+**Pattern 3: Polymorphic entities**
```

Comment:
I think that polymorphic entities belongs in their own chapter. Either we do that in this PR, or we write that down as a todo for later on.